autogpt/commands/web_selenium.py

```diff
@@ -66,23 +68,31 @@ def browse_website(url: str, question: str, agent: Agent) -> str:
     Returns:
         Tuple[str, WebDriver]: The answer and links to the user and the webdriver
     """
+    driver = None
     try:
         driver, text = scrape_text_with_selenium(url, agent)
     except WebDriverException as e:
         # These errors are often quite long and include lots of context.
         # Just grab the first line.
         msg = e.msg.split("\n")[0]
         return f"Error: {msg}"
+    else:
+        add_header(driver)
 
-    add_header(driver)
-    summary = summarize_memorize_webpage(url, text, question, agent, driver)
-    links = scrape_links_with_selenium(driver, url)
+        token_length = count_string_tokens(text, agent.llm.name)
+        if token_length > TOKENS_TO_TRIGGER_SUMMARY:
+            summary = summarize_memorize_webpage(url, text, question, agent, driver)
+            links = scrape_links_with_selenium(driver, url)
 
-    # Limit links to 5
-    if len(links) > 5:
-        links = links[:5]
-    close_browser(driver)
-    return f"Answer gathered from website: {summary}\n\nLinks: {links}"
+            # Limit links to 5
+            if len(links) > 5:
```

Comment:
Can you lift the max links to a constant as well?

Reply:
Yeah - I wonder whether we should also randomize the list? That could reduce loops.

Reply:
I don't think you want to randomize.  The link order contains information.